js/main.js

```diff
@@ -13,12 +13,11 @@ function switchView(dataView) {
 
 function switchViewing(event) {
   event.preventDefault();
-  var $viewSwapper = document.querySelectorAll('.view-swap');
-  for (var i = 0; i < $viewSwapper.length; i++) {
-    if ($viewSwapper[i] === event.target) {
-      var dataView = $viewSwapper[i].getAttribute('data-view');
-      switchView(dataView);
-    }
+  if (!event.target.matches('.view-swap')) {
+    return;
+  }
+  if (event.target.matches('.view-swap')) {
```

Comment:
You can remove this conditional and just call the function inside it directly. If you got past the previous conditional, that means the element does have the `view-swap` class, so you can just do the work. The magic of guards!!